docs/design/thee_datahall.md

```diff
@@ -0,0 +1,197 @@
+        nodeSelectorTerms:
+        - matchExpressions:
+          - key:  topology.kubernetes.io/zone 
+            operator: In
+            values:
+            - az1
+            - az2
+            - az3
+```
+
+The `NodeAffinity` must be set by the user and the operator doesn't take any action to automatically set the value.
+The user would only require to create one `FoundationDBCluster` and set the `redundancy_mode` to `three_data_hall` or `three_datacenter`.
+
+#### Multiple Kubernetes clusters
+
+In this deployment scenario a user would have 3 different Kubernetes clusters where each cluster spans across a different AZ.
+One requirement is that all Pods in the different AZs are able to communicate.
+The deployment pattern is similar to the [multi Kubernetes deployment](https://github.com/FoundationDB/fdb-kubernetes-operator/blob/master/docs/manual/fault_domains.md#option-2-multi-kubernetes-replication) but we can't use the special key here since that would mean that the `locality_zone` field would be set to the Kubernetes clusters zone ID and for `three_data_hall` or `three_datacenter` we need at least 2 zones per `data_hall`/`data_center`.
+The initial deployment would be splitt into two phases:
+
+1. Deploy the seed cluster into one of the Kubernetes clusters and wait until it's fully reconciled.
+1. Copy the connection string and use it as `seedConnectionString` for the other two clusters and set `redundancy_mode: "three_data_hall"`.
+
+The initial cluster spec could look like this:
+
+```yaml
+apiVersion: apps.foundationdb.org/v1beta1
+kind: FoundationDBCluster
+metadata:
+  labels:
+    cluster-group: test-cluster
+  name: test-cluster
+spec:
+  version: 6.2.30
+  instanceIDPrefix: az1
+  databaseConfiguration:
+    redundancy_mode: "triple"
+  localities:
+  - key: "locality_data_hall"
+    valueFromNode: "topology.kubernetes.io/zone"
+```
+
+With this configuration we will use the default fault domain the Kubernetes nodes as `locality_zone`.
+When the clutser is reconciled we can create the `FoundationDBCluster` spec in the other two Kubernetes clusters and set the `redundancy_mode: "three_data_hall"`:
+
+```yaml
+apiVersion: apps.foundationdb.org/v1beta1
+kind: FoundationDBCluster
+metadata:
+  labels:
+    cluster-group: test-cluster
+  name: test-cluster
+spec:
+  version: 6.2.30
+  instanceIDPrefix: $az
+  seedConnectionString: $connectionString
+  databaseConfiguration:
+    redundancy_mode: "three_data_hall"
+  localities:
+  - key: "locality_data_hall"
+    valueFromNode: "topology.kubernetes.io/zone"
+```
+
+The processes in the other Kubernetes clusters will join the current `FoundationDBCluster`.
+Once enough processes joined the cluster one of the operator will select 5 Coordinators span across 3 `data_halls`.
+
+### Coordinator selection
+
+The coordinator selecion must be adjusted to select at most 2 coordinators in the same `data_hall` or `datacenter` depending on the redundancy mode.
```

Comment:
```suggestion
The coordinator selection must be adjusted to select at most 2 coordinators in the same `data_hall` or `datacenter` depending on the redundancy mode.
```

Reply:
I think we want to use 9 coordinators in three data hall mode, which would mean we would be fine with up to 4 coordinators in a single data hall.

Reply:
That's fine for I just oriented at the FoundationDB docs saying 5 coordinators are recommended.